particles/Tutorial/Kotlin/common/Particle.kt

```diff
@@ -0,0 +1,327 @@
+package arcs
```

Comment:
We could do this in a separate PR, but this (and some files below) are as you note in the path `common`, so it seems like we should put them somewhere they can be shared so that other Kotlin particles can reference them without having to reach into `particles/Tutorial`?

And/or, should we move them into a layer in Runtime in some manner, like how we have `dom-particle.ts` which JS particle authors can just extend in a straightforward manner without having to have their own copy?

All of the gradle stuff as well... seems like we need to refactor somehow so we can have a small one that just references base gradle config/logic/etc.

Reply:
Agreed. We also found some issues with the schema2pkg when JSON is involved. I haven't included the fixes in this PR for simplicity.

Reply:
I'm currently working on a gradle build rule to compile Particle.kt and the interop as separate binaries from each Wasm executable. 

`Particle.kt` and the interop are currently in src/wasm/kotlin.

I've been working through the docs for Gradle wrt multi-project builds, too.

This is an early version of the branch; we intend to change it a lot before it's out of WIP.